internal/buildpack/locator_type.go

```diff
@@ -74,8 +81,14 @@ func GetLocatorType(locator string, buildpacksFromBuilder []dist.BuildpackInfo)
 		return IDLocator, nil
 	}
 
-	if _, err := name.ParseReference(locator); err == nil {
-		return PackageLocator, nil
+	if strings.Contains(locator, "@sha") || strings.Count(locator, "/") > 1 {
+		if _, err := name.ParseReference(locator); err == nil {
+			return PackageLocator, nil
+		}
+	}
+
+	if strings.Count(locator, "/") == 1 {
+		return RegistryLocator, nil
 	}
```

Comment:
If this is all trying to parse a naked locator, can we pull it out to a separate func, to make that a bit more clear?

Reply:
addressed in d127f425